src/GoogleFit/FitOnFhir.GoogleFit/Services/GoogleFitImportService.cs

```diff
@@ -165,9 +167,10 @@ private string GenerateDataSetId(long lastSyncTimeNanos)
         {
             if (lastSyncTimeNanos == default)
             {
-                // if this DataSource has never been synced before, then retrieve 30 days prior worth of data
-                // and convert to DateTimeOffset to so .NET unix conversion is usable
-                DateTimeOffset startDateDto = _utcNowFunc().AddDays(-30);
+                // if this DataSource has never been synced before, then retrieve an amount of prior data
+                // as determined by the value in 'HistoricalImportTimeSpan' and convert to DateTimeOffset
+                // so .NET unix conversion is usable
+                DateTimeOffset startDateDto = _utcNowFunc().Subtract(_options.HistoricalImportTimeSpan);
```

Comment:
I wonder if we should check if _options.HistoricalImportTimeSpan is default, or if the resulting `startDateDto` is BEFORE `endDate`. I believe things like -1.00:00:00 will actually work (making the DataSetId invalid) or 0.00:00:00, which would mean "do not collect any historical data.

I think in these 2 cases, I think we should set the start date == the end date. In other words, if start date is greater than end date, start date should be set to end date.